src/main/java/com/ibm/crypto/plus/provider/ock/NativeInterface.java

```diff
@@ -131,140 +131,50 @@ static void preloadJGskit(String libraryToLoad) {
         osArch = System.getProperty("os.arch");
         String jgskitPath = getJGskitLoadPath();
         File loadFile = null;
-        if (osName.startsWith("Windows")) {
-            if (osArch.equals("x86")) {
-                // win32_x86
-                loadFile = new File(jgskitPath, "libjgskit.dll");
-            } else {
-                // win64_x86
-                loadFile = new File(jgskitPath, "libjgskit_64.dll");
-            }
+        if (osName.startsWith("Windows") && osArch.equals("amd64")) {
+            loadFile = new File(jgskitPath, "libjgskit_64.dll");
         } else if (osName.equals("Mac OS X")) {
             loadFile = new File(jgskitPath, "libjgskit.dylib");
         } else {
             loadFile = new File(jgskitPath, "libjgskit.so");
         }
 
-
         boolean jgskitLibraryPreloaded = loadIfExists(loadFile);
         if (jgskitLibraryPreloaded == false) {
-            String exceptionMessage = "Could not load dependent jgskit library";
-
-            if (debug != null) {
-                // Do not use loadFile or libraryName in message in an effort to hide OCK usage
-                // from users
-                //
-                exceptionMessage = "Could not load dependent jgskit library for os.name=" + osName
-                        + ", os.arch=" + osArch;
-            }
-
-            throw new ProviderException(exceptionMessage);
+            throw new ProviderException("Could not load dependent jgskit library for os.name=" + osName
+                        + ", os.arch=" + osArch);
         }
     }
 
-    static void preloadOCK(String libraryToLoad, boolean add64) {
+    static void preloadOCK(String libraryToLoad) {
```

Comment:
I'm not even sure why we pass `libraryToLoad` as a parameter. Is the method ever called with a different option?

Reply:
Good point. Updated the code here to use a constant since we always pass the same value into this method. The argument was completely removed.

Reply:
Updated with https://github.com/IBM/OpenJCEPlus/pull/348/commits/97153a9c2dcb73771278d1c69605df14dd31a0b4